isovar/reads_at_locus.py

```diff
@@ -0,0 +1,210 @@
+# Copyright (c) 2016. Mount Sinai School of Medicine
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module wraps pysam and gives us a view of any reads overlapping
+a variant locus which includes offsets into the read sequence & qualities
+for extracting variant nucleotides.
+"""
+
+from __future__ import print_function, division, absolute_import
+
+from collections import namedtuple
+
+from .logging import create_logger
+
+logger = create_logger(__name__)
+
+ReadAtLocus = namedtuple(
+    "ReadAtLocus",
+    [
+        "name",
+        "sequence",
+        "reference_positions",
+        "quality_scores",
+        "base0_read_position_before_variant",
+        "base0_read_position_after_variant",
+    ])
+
+DEFAULT_MIN_MAPPING_QUALITY = 1
+DEFAULT_USE_DUPLICATE_READS = False
+DEFAULT_USE_SECONDARY_ALIGNMENTS = True
+
+def gather_reads_at_locus(
+        samfile,
+        chromosome,
+        base1_position_before_variant,
+        base1_position_after_variant,
+        use_duplicate_reads=DEFAULT_USE_DUPLICATE_READS,
+        use_secondary_alignments=DEFAULT_USE_SECONDARY_ALIGNMENTS,
+        min_mapping_quality=DEFAULT_MIN_MAPPING_QUALITY):
+    """
+    Generator that yields a sequence of ReadAtLocus records for reads which
+    contain the positions before and after a variant. The actual work to figure
+    out if what's between those positions matches a variant happens later in
+    the `variant_reads` module.
+
+    Parameters
+    ----------
+    samfile : pysam.AlignmentFile
+
+    chromosome : str
+
+    base1_position_before_variant : int
```

Comment:
Just poking around; noticed that you have two `base1_position_before_variants`s vs. `after`.